src/System.Collections/src/System/Collections/Generic/SortedSet.cs

```diff
@@ -1132,13 +1132,14 @@ internal virtual void IntersectWithEnumerable(IEnumerable<T> other)
                 if (Contains(item))
                 {
                     toSave.Add(item);
+                    DoRemove(item);
```

Comment:
No, the test included fails without the removal of the `if (toSave.Count < Count)`, but if you simply delete that line from the original source, the test you've added passes.